### PR TITLE
Add `Distributive` and `Representable` instances to `Vec n`

### PR DESCRIPTION
**Pull request by gergoerdi**

### CODE REVIEW

File: clash-prelude/clash-prelude.cabal [hunk 1]
```diff
@@ -317,7 +317,8 @@ Library
                       TypeFamilies
                       UndecidableInstances
 
-  Build-depends:      array                     >= 0.5.1.0 && < 0.6,
+  Build-depends:      adjunctions               >= 4.0,
```
Comment (christiaanb) [inline clash-prelude/clash-prelude.cabal:320]: Could you add an upper bound `< 5`

File: clash-prelude/clash-prelude.cabal [hunk 2]
```diff
@@ -327,6 +328,7 @@ Library
                       data-binary-ieee754       >= 0.4.4   && < 0.6,
                       data-default-class        >= 0.1.2   && < 0.2,
                       deepseq                   >= 1.4.1.0 && < 1.6,
+                      distributive              >= 0.1,
```
Comment (christiaanb) [inline clash-prelude/clash-prelude.cabal:331]: Could you add an upper bound `< 1.0`

**Review by christiaanb — APPROVED** [review]
Looks good. Could you add a changelog entry? See https://github.com/clash-lang/clash-compiler/blob/master/changelog/README.md

**Comment by gergoerdi** [comment]
ping?

**Comment by DigitalBrains1** [comment]
@kloonbot run_ci f5be71d

**Comment by DigitalBrains1** [comment]
ICMP Echo Reply ;-)

**Pull request merged**